### PR TITLE
Backport pr 585

### PR DESCRIPTION
**Pull request by ncabatoff**

### CODE REVIEW

File: freelist_hmap.go [hunk 1]
```diff
@@ -1,10 +1,21 @@
 package bbolt
 
-import "sort"
+import (
+	"go.etcd.io/bbolt/internal/common"
+	"sort"
```
Comment (raskchanky) [inline freelist_hmap.go:4]: I'm not certain how picky we want to be with stuff like the imports, given that it seems like this is a temporary repo, but this doesn't match our normal style.

**Review by raskchanky — APPROVED** [review]
I left one tiny nit but nothing to prevent merging.

I also notice that the changes in the original PR to `freelist_test.go` and `internal/common/utils.go` were not brought over. Is that on purpose?

CleanShot 2023-11-03 at 09 40 05

Other than that though, looks great!

**Comment by ncabatoff** [comment]
> I also notice that the changes in the original PR to `freelist_test.go` and `internal/common/utils.go` were not brought over. Is that on purpose?

If you look on the release-1.3 branch we're targeting, the test in question and the utils.go file are both absent.

**Comment by raskchanky** [comment]
@ncabatoff Makes sense, thanks for clarifying.

File: freelist_hmap.go (hunk 1 above)
Comment (ncabatoff) [reply freelist_hmap.go:4]: Fixed.

**Pull request merged**